com.rockwellcollins.atc.resolute.analysis/src/com/rockwellcollins/atc/resolute/analysis/values/ResoluteRecordValue.java

```diff
@@ -10,36 +10,46 @@
 
 public class ResoluteRecordValue extends ResoluteValue {
 	private final Map<String, ? extends ResoluteValue> fields;
-	
+
 	public ResoluteRecordValue(Map<String, ? extends ResoluteValue> fields) {
 		if (fields == null) {
 			throw new IllegalArgumentException("Cannot create null ResoluteRecordValue");
 		}
-		
+
 		this.fields = Collections.unmodifiableMap(new HashMap<>(fields));
 	}
-	
+
 	public ResoluteValue getField(String fieldName) {
 		return fields.get(fieldName);
 	}
-	
+
+	@Override
+	public boolean isRecord() {
+		return true;
+	}
+
+	@Override
+	public ResoluteRecordValue getRecord() {
+		return this;
+	}
+
 	@Override
 	public ResoluteType getType() {
 		return BaseType.RECORD;
```

Comment:
As written, ResoluteRecordValues all share the same base type.  Comparison between the types two different records (records with differing fields) will yield a positive result because only the base type is returned.  To determine whether the types of two record values are equivalent the user must manually iterate through the fields checking name/type pairs.  This is even a little more hairy for record fields that are themselves records.  Is this the behavior we want?

Reply:
For now, yes.  Since Resolute doesn't provide the ability to define the structure of a record, this will not be an issue.  But if this capability is provided in the future we will need to revisit this.